go/vt/vtgate/router.go

```diff
@@ -357,47 +364,35 @@ func (rtr *Router) execInsertUnsharded(vcursor *requestContext, route *engine.Ro
 }
 
 func (rtr *Router) execInsertSharded(vcursor *requestContext, route *engine.Route) (*sqltypes.Result, error) {
-	var firstKsid []byte
-	inputs := route.Values.([]interface{})
-
 	insertid, err := rtr.handleGenerate(vcursor, route.Generate)
 	if err != nil {
 		return nil, fmt.Errorf("execInsertSharded: %v", err)
 	}
+	keyspace, routing, shardKsidMap, err := rtr.getInsertShardedRoute(vcursor, route)
```

Comment:
I think this can be simplified by combining it with rewriteMultiRowQuery, where this one step returns the `Mid` strings instead of routing. We could actually push this one step further to directly return shardQueries (instead of routing & shardKsidMap). We'll need to look at the code to see if that would make sense.